### DISCUSSION TITLE
why regionManager.Regions.ContainsRegionWithName("ContentRegion") return false;

### DISCUSSION THREAD

**Discussion by pengqi1991** [PrismLibrary/Prism#2327] [WPF]
why regionManager.Regions.ContainsRegionWithName("ContentRegion") return false;
_regionManager.RequestNavigate(RegionNames.ContentRegion, navigationPath); can't show content.
code in https://github.com/pengqi1991/scoremanager

**Comment by brianlagunas** [comment]
Can you place your sample on GitHub or use a .zip format?

This usually happens when you are trying to access the region before it exists, or when you define region on an element that is not part of the visual tree like a context menu.

**Reply by pengqi1991** [reply]
i have upload the sample

**Reply by brianlagunas** [reply]
It's exactly as I said. You are defining a region inside of an element that is not pat of the visual tree. The syncfusion NavDrawer is your problem.

```
            <syncfusion:SfNavigationDrawer.ContentView>
                <ContentControl prism:RegionManager.RegionName="{x:Static core:RegionNames.ContentRegion}" />
            </syncfusion:SfNavigationDrawer.ContentView>
```
You'll need to set the region name and the region manager in code-behind. GIve the ContentControl an x:Name and the do this in the code behind:
```
            RegionManager.SetRegionName(_contentRegion, Core.RegionNames.ContentRegion);
            RegionManager.SetRegionManager(_contentRegion, regionManager);
```

**Reply by pengqi1991** [reply]
thank you every much.it's mean don't need prism:RegionManager.RegionName="{x:Static core:RegionNames.ContentRegion}"?
how to know the usercontrol not pat of the visual tree.
